vexcl/optimisation.hpp

```diff
@@ -0,0 +1,200 @@
+#ifndef OPTIMISATIONS_HPP
+#define OPTIMISATIONS_HPP
+
+
+#include <vexcl/operations.hpp>
+
+namespace vex {
+//_________________________________________________ restrict keyword
```

Comment:
Can you please enclose the supporting code (that is not indended for direct user consumption) between
```
/// \cond INTERNAL
```
and 
```
/// \endcond
```
comments? See [tagged_terminal.hpp](https://github.com/ddemidov/vexcl/blob/7ac0b2516bc3317abf9893a5afb0828a05a93c6d/vexcl/tagged_terminal.hpp#L44) for an example. This would help Doxygen to  skip unneeded code during documentation generation.